lib/crawler/api/config.rb

```diff
@@ -339,7 +334,23 @@ def configure_sitemap_urls!
         end
       end
 
-      #---------------------------------------------------------------------------------------------
+      def configure_extraction_rules!
+        @extraction_rules = @domains.each_with_object({}) do |domain, extraction_rules|
+          url = domain[:url]
+          rulesets = domain[:extraction_rulesets].nil? ? [] : domain[:extraction_rulesets]
+
+          raise ArgumentError, "Extraction rulesets for #{url} is not an array" unless rulesets.is_a?(Array)
+
+          extra_rules = rulesets.flat_map(&:keys) - EXTRACTION_RULES_FIELDS
```

Comment:
I was surprised that ruby set math will remove more than one instance.
`[:url_filters, :rules, :url_filters, :rules] - [:url_filters, :rules] == []` 🤯